core/src/com/unciv/ui/options/DisplayTab.kt

```diff
@@ -94,6 +96,25 @@ private fun addMinimapSizeSlider(table: Table, settings: GameSettings, selectBox
     table.add(minimapSlider).minWidth(selectBoxMinWidth).pad(10f).row()
 }
 
+private fun addUnitIconAlphaSlider(table: Table, settings: GameSettings, selectBoxMinWidth: Float) {
+    table.add("Unit icon opacity".toLabel()).left().fillX()
+
+    val getTipText: (Float) -> String = {"%.0f".format(it*100) + "%"}
+
+    val unitIconAlphaSlider = UncivSlider(
+        0f, 1f, 0.1f, initial = settings.unitIconOpacity, getTipText = getTipText
+    ) {
+        settings.unitIconOpacity = it
+        settings.save()
+
+        val worldScreen = UncivGame.Current.getWorldScreenIfActive()
+        if (worldScreen != null)
+            worldScreen.shouldUpdate = true
```

Comment:
this would be equivalent to `worldscreen?.shouldUpdate =  true`. Just FYI, no need to change this now